middlewares/replace_path_regex.go

```diff
@@ -0,0 +1,38 @@
+package middlewares
+
+import (
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/containous/traefik/log"
+)
+
+// ReplacePathRegex is a middleware used to replace the path of a URL request with a regular expression
+type ReplacePathRegex struct {
+	Handler     http.Handler
+	Regexp      *regexp.Regexp
+	Replacement string
+}
+
+// NewReplacePathRegexHandler returns a new instance of ReplacePathRegex
+func NewReplacePathRegexHandler(regex string, replacement string, handler http.Handler) http.Handler {
+	exp, err := regexp.Compile(strings.TrimSpace(regex)) // `exp` will be nil when error
+	if err != nil {
+		log.Warnf("Error compiling regular expression %s: %s", regex, err)
```

Comment:
in this case `exp` is nil, you cannot use `log.Warnf` you must use `log.Errorf`.